# HG changeset patch
# Parent  53fa843ad516a35dd7b58071f5c7642e0c5c7601
Fix LIFX addressing whenever multiple bulbs are involved

Looks like the "addressable" bit in the "protocol" header field has to
always be true.

diff --git a/lifx/wire_proto.c b/lifx/wire_proto.c
--- a/lifx/wire_proto.c
+++ b/lifx/wire_proto.c
@@ -226,19 +226,19 @@
 
     switch (target_type) {
     case LGTD_LIFX_TARGET_SITE:
+    case LGTD_LIFX_TARGET_ALL_DEVICES:
         hdr->protocol.tagged = true;
+        hdr->protocol.addressable = true;
         break;
     case LGTD_LIFX_TARGET_TAGS:
         hdr->protocol.tagged = true;
+        hdr->protocol.addressable = true;
         hdr->target.tags = target.tags;
         break;
     case LGTD_LIFX_TARGET_DEVICE:
         hdr->protocol.addressable = true;
         memcpy(hdr->target.device_addr, target.addr, LGTD_LIFX_ADDR_LENGTH);
         break;
-    case LGTD_LIFX_TARGET_ALL_DEVICES:
-        hdr->protocol.tagged = true;
-        break;
     }
 
     lgtd_lifx_wire_encode_header(hdr);
